src/gui/toolbarMenubar/model/ToolbarColorNames.cpp

```diff
@@ -1,50 +1,26 @@
 #include "ToolbarColorNames.h"
 
 #include <fstream>
+#include <memory>
 
 #include <glib/gstdio.h>
+#include <util/PathUtil.h>
 
 #include "i18n.h"
 
-ToolbarColorNames::ToolbarColorNames() {
-    this->config = g_key_file_new();
-    g_key_file_set_string(this->config, "info", "about", "Xournalpp custom color names");
-    initPredefinedColors();
-}
-
-ToolbarColorNames::~ToolbarColorNames() {
-    g_key_file_free(this->config);
-}
-
-static ToolbarColorNames* instance = nullptr;
-
-auto ToolbarColorNames::getInstance() -> ToolbarColorNames& {
-    if (instance == nullptr) {
-        instance = new ToolbarColorNames();
-    }
-
-    return *instance;
-}
-
-void ToolbarColorNames::freeInstance() {
-    delete instance;
-    instance = nullptr;
-}
-
-void ToolbarColorNames::loadFile(fs::path const& file) {
+void load(fs::path const& file, GKeyFile* config, std::unordered_map<Color, std::string>& predefinedColorNames) {
     GError* error = nullptr;
     if (!g_key_file_load_from_file(config, file.u8string().c_str(), G_KEY_FILE_NONE, &error)) {
         g_warning("Failed to load \"colornames.ini\" (%s): %s\n", file.string().c_str(), error->message);
         g_error_free(error);
         return;
     }
-
-    g_key_file_set_string(this->config, "info", "about", "Xournalpp custom color names");
+    g_key_file_set_string(config, "info", "about", "Xournalpp custom color names");
 }
 
-void ToolbarColorNames::saveFile(fs::path const& file) {
+void safe(fs::path const& file, GKeyFile* config, std::unordered_map<Color, std::string>& predefinedColorNames) {
```

Comment:
Typo, should be `save`. Also needs to be marked `static`.

Reply:
> Also needs to be marked `static`.

The function is already static.

Reply:
renamed and namespaced.